Dockerfile

```diff
@@ -0,0 +1,16 @@
+FROM golang:1.22.3-alpine
```

Comment:
Optimisations that I can think of:

- multi staged builds: first stage can use the `golang` image to build the go binary. second stage can be a minimal image like alpine that only includes the built binary, this should reduce the final image size.

Reply:
- layers reductions